src/Stratis.Bitcoin.Features.Consensus/Rules/ConsensusRule.cs

```diff
@@ -45,12 +39,51 @@ public virtual void Initialize()
     }
 
     /// <summary>
-    /// Rules that are manipulating state.
+    /// Provide additional information about a consensus rule that can be used by the rule engine.
     /// </summary>
-    public abstract class ExecutionConsensusRule : ConsensusRule
+    public class ConsensusRuleDescriptor
```

Comment:
This class can easily be split into multiple classes.

Reply:
Sure I will do it in the next PR for rules.